### ISSUE TITLE
`convert_module_paths(idl: Idl)` is too naive

### ISSUE THREAD

**Issue by blackbeam** [coral-xyz/anchor#3358]
See [here](https://github.com/coral-xyz/anchor/blob/master/idl/src/build.rs#L304)

Consider the following code. The docstring for the constant will find it's way into the generated IDL and later will confuse the `convert_module_paths` triggering false-positive for `has_conflict`:

```rust
#[account]
struct Foo {
    // ...
}

// and somewhere else

/// Seed for [`Foo`]
///
/// [`Foo`]: crate::some_mod::Foo
#[constant]
pub const FOO_SEED: &[u8] = b"FOO";
```

**Comment by acheroncrypto** [comment]
Yeah, we probably need to discard doc comments before doing the conversion, but you should also be able to avoid triggering this issue quite easyily. The reason this happens is because `crate::some_mod::Foo` is the last part of the doc comment. Even if you add a single space at the end, it should work because we're specifically checking for `::{name}"` (notice the `"` at the end):

https://github.com/coral-xyz/anchor/blob/e4f5d43a062a408213e6e47a79088dec847fc6e7/idl/src/build.rs#L317

Basically any combination where the full path is not specified exactly at the end should work:

```rs
/// Seed for [`crate::some_mod::Foo`]
#[constant]
pub const FOO_SEED: &[u8] = b"FOO";
```
